index.js

```diff
@@ -1,42 +1,36 @@
-const once = require('once');
-const { Readable, Transform } = require('stream');
-module.exports = function (fn, state = {}, opts) {
-  let finished = false;
-
-  let interval = opts.interval || 1000;
-  let streamOpts = Object.assign(
-    { objectMode: true, interval: undefined },
-    opts,
-    { read: once(poll) });
-
-  const rs = Readable(streamOpts);
+const { PassThrough } = require('stream');
 
-  function poll() {
-    const batchStream = fn(state);
+class BatchStream extends PassThrough {
```

Comment:
Call this `PollingStream` instead of `BatchStream` as the batch stream is the individual batches that get streamed between polls.